pkg/helm/v3/converter.go

```diff
@@ -0,0 +1,64 @@
+package v3
+
+import (
+	"github.com/helm/helm-2to3/pkg/common"
+	helm2to3v2 "github.com/helm/helm-2to3/pkg/v2"
+	helm2to3v3 "github.com/helm/helm-2to3/pkg/v3"
+)
+
+// Converter Converts a given helm 2 release with all its release versions to helm 3 format and deletes the old release from tiller
+type Converter struct {
+	TillerNamespace string
+	KubeConfig      string // file path to kubeconfig
+}
+
+// V2ReleaseExists helps you check if a helm v2 release exists or not
+func (c Converter) V2ReleaseExists(releaseName string) (bool, error) {
+	retrieveOpts := helm2to3v2.RetrieveOptions{
+		ReleaseName:     releaseName,
+		TillerNamespace: c.TillerNamespace,
+	}
+	kubeConfig := common.KubeConfig{
+		File: c.KubeConfig,
+	}
+	v2Releases, err := helm2to3v2.GetReleaseVersions(retrieveOpts, kubeConfig)
+	if err != nil {
+		return false, err
+	}
+	return len(v2Releases) > 0, nil
+}
+
+// Convert attempts to convert the given release name from v2 to v3.
+func (c Converter) Convert(releaseName string, dryRun bool) error {
+	retrieveOpts := helm2to3v2.RetrieveOptions{
+		ReleaseName:     releaseName,
+		TillerNamespace: c.TillerNamespace,
+	}
+	kubeConfig := common.KubeConfig{
+		File: c.KubeConfig,
+	}
+	v2Releases, err := helm2to3v2.GetReleaseVersions(retrieveOpts, kubeConfig)
+	if err != nil {
+		return err
+	}
+
+	relCount, start := len(v2Releases), 0
+
+	if !dryRun {
+		for i := start; i < relCount; i++ {
```

Comment:
nit: why not just use `i := 0` instead of introducing `start`? (or use `for _, v2Release := range v2Releases`)

Reply:
I do not think you can get away with just `range` as this does not guarantee order which is a requirement for this import.

Reply:
TIL... I'll revert back to a regular for loop

Reply:
@hiddeco `helm2.GetReleaseVersions()` returns a slice, why would it not guarantee order?

Reply:
I just checked and it is indeed a slice `[]*rls.Release`. Everything I'm reading online points to slices having order guarantees across the board. I'll switch back to range approach. Also, I need to trigger another e2e test.